### ISSUE TITLE
Clean isEven, isOdd and such 

### ISSUE THREAD

**Issue by Breush** [linbox-team/linbox#137]
Following https://github.com/linbox-team/linbox/pull/134#discussion_r218097621, we should make overloads for `isEven(size_t)` and such.

I vote for a generic:
```c++
template<typename IntType>
bool isOdd(const IntType& value) { return value & IntType(1); }
```
With some specializations if needed.

__NOTE__: In Givaro, the only definition of `isOdd` is for `Integer` (and `isEven` does not exist). That means that we shouldn't use the Givaro one in LinBox.

__NOTE 2__: In `linbox/integer.h` there is also `isPositive` and `IsNegative` (notice the capital for the negative one) which should be cleaned.

**Comment by david-lucas** [comment]
See PR for #155.

**Comment by ClementPernet** [comment]
Fixed in #155
